### ISSUE TITLE
numpy.random.permutation modified pandas.RangeIndex

### ISSUE THREAD

**Issue by magicalpear** [numpy/numpy#11975]
### Reproducing code example:

```python
import numpy as np
import pandas as pd

ri = pd.RangeIndex(2)
print('before permutation:', ri.values)

while True:
    idx = np.random.permutation(ri)
    if idx[0] == 1:
        break
    
print('after permutation:', ri.values)
```

### Error message:
Running the code outputs the following:

before permutation: [0 1]
after permutation: [1 0]

np.random.permutation should not change input based on manual.

The code behavior correctly if change the code to

    idx = np.random.permutation(ri.values)

### Numpy/Python version information:

1.15.1 3.6.5 |Anaconda, Inc.| (default, Mar 29 2018, 13:32:41) [MSC v.1900 64 bit (AMD64)]
``

**Comment by abhinuvpitale** [comment]
I dont understand whats the error here, shouldn't that be the right behavior?
What do you mean by manual?

**Comment by abhinuvpitale** [comment]
This is because when you call ```np.random.permutation```, the call goes to [here](https://github.com/numpy/numpy/blob/master/numpy/core/numeric.py#L433), where the RangeIndex is first converted to an array!

**Comment by magicalpear** [comment]
I agree. RangeIndex should be converted to an array and then permutation is on the new array. The original RangeIndex should not have been changed.

**Comment by bashtage** [comment]
This line

https://github.com/numpy/numpy/blob/master/numpy/random/mtrand/mtrand.pyx#L4911

should strictly check for ndarray since the is is failing but the asarray must be sharing memory somehow.

Also happens with `Series` and probably most other 1d arrays that don't return a copy when called with asarray.

**Comment by bashtage** [comment]
Might be able to use `may_share_memory` as a better check.

**Comment by mattip** [comment]
`np.may_share_memory(a, ri)` is True, so it should work. Also needs clarification in the documentation when the operation is in-place

**Comment by bashtage** [comment]
I don't think it should every be in-place. shuffles should always be on copies or new arrays.

**Comment by boldt** [comment]
Will it land in 1.15.3 or later in 1.16.*? Ot has side effects in tensorsflow (see my referenced bugs before)?

**Comment by charris** [comment]
It is scheduled for backport to 1.15.3.

**Comment by boldt** [comment]
Thanks. One last question: Since I am new to the python world and thus also new to numpy: do you have a release plan? Or do you have already a date for 1.15.3?

**Comment by charris** [comment]
1.15.3 is planned for about a week from now, although our release software has been broken by incompatible upstream changes, so there may be a delay.
